.github/workflows/reusable-e2e.yml

```diff
@@ -82,6 +90,13 @@ jobs:
         with:
           path: ~/.cache/Cypress
           key: ${{ runner.os }}-cypress-${{ hashFiles('yarn.lock') }}
+      - name: Cache Puppeteer runner
+        # create new puppeteer cache entry only if puppeteer cache missed and we installed a new one.
+        if: steps.restore-puppeteer-cache.outputs.cache-hit != 'true'
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/puppeteer
+          key: ${{ runner.os }}-puppeteer-${{ hashFiles('yarn.lock') }}
```

Comment:
This step can actually be removed, because "Restore Puppeteer runner from cache" step above will automatically upload and populate cache if the cache missed. 

>  If no cache exactly matches the provided key, this is considered a cache miss. On a cache miss, the action automatically creates a new cache if the job completes successfully.
> - https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#using-the-cache-action

So we can remove this step here in favor of the step above at line 76.

Reply:
This is on me, because I misunderstood how GHA cache works when I first did the cypress caching that you referenced 😅

Reply:
Good point. I was wondering about that. Done 628100a81344d799128d93846ddb3eb9f7aa11aa

Do you think I should remove the cypress caching step in this PR?

Reply:
I think that should be a separate PR just to keep this PR focused on link checker 👍 

I'd appreciate a new PR though!

Reply:
Re-done the commit 92bd0791074a87afcb55af09af748e46fef248de